docs/content/Switch.mdx

```diff
@@ -0,0 +1,225 @@
+</Box>
+```
+
+### Controlled
+
+```javascript noinline live
+const Controlled = () => {
+  const [isOn, setIsOn] = React.useState(false)
+
+  const onClick = () => {
+    setIsOn(!isOn)
+  }
+
+  const handleSwitchChange = on => {
+    console.log(`new switch "on" state: ${on}`)
+  }
+
+  return (
+    <>
+      <Box display="flex" maxWidth="300px">
+        <Box flexGrow={1} fontSize={2} fontWeight="bold" id="switchLabel">
+          Notifications
+        </Box>
+        <Switch onClick={onClick} onChange={handleSwitchChange} on={isOn} aria-labelledby="switchLabel" />
+      </Box>
+      <p>The switch is {isOn ? 'on' : 'off'}</p>
+    </>
+  )
+}
+
+render(Controlled)
+```
+
+### Small
+
+```jsx live
+<Box display="flex" maxWidth="300px">
+  <Box flexGrow={1} fontSize={1} fontWeight="bold" id="switchLabel">
+    Notifications
+  </Box>
+  <Switch aria-labelledby="switchLabel" size="small" />
+</Box>
+```
+
+### Delayed toggle with loading state
+
+```javascript noinline live
+const LoadingToggle = () => {
+  const [isLoading, setIsLoading] = React.useState(false)
+  const [isOn, setIsOn] = React.useState(false)
+
+  async function switchSlowly(currentOn) {
+    await new Promise(resolve => setTimeout(resolve, 1500))
+    return await !currentOn
+  }
+
+  async function onClick() {
+    setIsLoading(!isLoading)
+    const newSwitchState = await switchSlowly(isOn)
+    setIsOn(newSwitchState)
+  }
+
+  const handleSwitchChange = React.useCallback(
+    on => {
+      setIsLoading(false)
+    },
+    [setIsLoading]
+  )
+
+  return (
+    <>
+      <Box display="flex" maxWidth="300px">
+        <Box flexGrow={1} fontSize={2} fontWeight="bold" id="switchLabel">
+          Notifications
+        </Box>
+        <Switch
+          aria-labelledby="switchLabel"
+          isLoading={isLoading}
+          on={isOn}
+          onClick={onClick}
+          onChange={handleSwitchChange}
+        />
+      </Box>
+      <p>The switch is {isOn ? 'on' : 'off'}</p>
+    </>
+  )
+}
+
+render(LoadingToggle)
+```
+
+### Disabled
+
+```jsx live
+<Box display="flex" maxWidth="300px">
+  <Box flexGrow={1} fontSize={2} fontWeight="bold" id="switchLabel">
+    Notifications
+  </Box>
+  <Switch aria-labelledby="switchLabel" disabled />
+</Box>
+```
+
+### With associated caption text
+
+```jsx live
+<Box display="flex">
+  <Box flexGrow={1}>
+    <Text fontSize={2} fontWeight="bold" id="switchLabel" display="block">
+      Notifications
+    </Text>
+    <Text color="fg.subtle" fontSize={1} id="switchCaption" display="block">
+      Notifications will be delivered via email and the GitHub notification center
+    </Text>
+  </Box>
+  <Switch aria-labelledby="switchLabel" aria-describedby="switchCaption" />
+</Box>
+```
+
+### Left-aligned with label
+
+```jsx live
+<>
+  <Text fontSize={2} fontWeight="bold" id="switchLabel" display="block" mb={1}>
+    Notifications
+  </Text>
+  <Switch statusLabelPosition="end" aria-labelledby="switchLabel" />
+</>
+```
```

Comment:
Are people allowed to render the toggle with a text label ('on' or 'off')?

Reply:
I don't understand what you mean. There's already an "on" and "off" label.

Reply:
Ah, I meant "without" 🤦 

Are people allowed to render the toggle _without_ a text label ('on' or 'off')?

Reply:
No - we intentionally don't allow a switch without the "on" or "off" label.